labs/b9.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Lab 9 - Version Control and Backups
+layout: lab
+---
+
+## Git IRL
+
+### Fixing mistakes
+If you ever feel like you need to start over, you can run these commands:
+```
+git checkout master
+git reset --hard b9_start
+```
+And if you created the branch `my_branch`, delete it with
+```
+git branch -D my_branch
+```
+
+### Pulling the code
+To clone the decal-lab git repository, run:
+``git clone https://github.com/0xcf/decal-labs.git``
+
+If you have already cloned the repository, go to your `decal-labs` directory
+and run `git pull` to pull the latest changes. The materials for this part of
+the lab will be in the `b9` directory.
+
+Also if you dont have `make` installed, run the following command:
+``sudo apt install make``
+
+### Making changes
+When making modifications to a repository, the best practice is to create a new
+branch for your changes. To do this, run `git checkout -b my_branch`. This
+command will create a new branch and switch to it.
+
+Now we will make a modification to some\_code.c and commit it. Change line 4 to
+`char *x = "Correct output";` and save. Now run
+```
+make
+./some_code
+```
+This should output `Correct output` on the console. Now stage your changes and
+make a new commit with a message saying "Added Correct output"
+(hint: [lecture slide 6](https://docs.google.com/presentation/d/1NxdvuLkWSzRvtnj41AkvGU00ub_XUbk8wlIxWxm0ok4/edit#slide=id.g707efdc427_0_0)).
+
+Now we're going to add another feature to some\_code.c. Under the comment on
+line 7, add the following code:
+```
+char* feature = "Heres my new feature!";
+puts(feature);
+```
+Now again run
+```
+make
+./some_code
+```
+And check that the output is "Correct output" followed by "Heres my new
+feature!". Now stage and commit these changes with the message "Added my new
+feature".
```

Comment:
We may want to give some hints about the commands to use here.

Reply:
Theres a hint above where they stage and commit other files